graph/import.go

```diff
@@ -19,21 +24,40 @@ func (s *TagStore) CmdImport(job *engine.Job) engine.Status {
 		return job.Errorf("Usage: %s SRC REPO [TAG]", job.Name)
 	}
 	var (
-		src          = job.Args[0]
-		repo         = job.Args[1]
-		tag          string
-		sf           = utils.NewStreamFormatter(job.GetenvBool("json"))
-		archive      archive.ArchiveReader
-		resp         *http.Response
-		stdoutBuffer = bytes.NewBuffer(nil)
-		newConfig    runconfig.Config
+		src           = job.Args[0]
+		repo          = job.Args[1]
+		tag           string
+		sf            = utils.NewStreamFormatter(job.GetenvBool("json"))
+		archiveReader archive.ArchiveReader
+		resp          *http.Response
+		stdoutBuffer  = bytes.NewBuffer(nil)
+		newConfig     runconfig.Config
 	)
 	if len(job.Args) > 2 {
 		tag = job.Args[2]
 	}
 
 	if src == "-" {
-		archive = job.Stdin
+		buf, _ := ioutil.ReadAll(job.Stdin)
+		archiveContent := bytes.NewBuffer(buf)
+		cannotImport := `Looks like you are trying to import a previously saved image. Did you mean to use docker load insted?`
+		ErrCannotImport := errors.New(cannotImport)
+
+		tmpDir, err := ioutil.TempDir("", "docker-import-")
+		if err != nil {
+			return job.Error(err)
+		}
+		defer os.RemoveAll(tmpDir)
+
+		if err := chrootarchive.Untar(archiveContent, tmpDir, &archive.TarOptions{}); err != nil {
```

Comment:
There's no point in decompressing the tarball. It can be checked by saving it to a temporary location and going through the entries in it.